lib/bolt/error.rb

```diff
@@ -25,20 +25,26 @@ def to_h
     def to_json(opts = nil)
       to_h.to_json(opts)
     end
+
+    def to_puppet_error
+      Puppet::DataTypes::Error.from_asserted_hash(to_h)
+    end
   end
 
   class RunFailure < Error
-    attr_reader :resultset
+    attr_reader :result_set
 
-    def initialize(resultset, action, object)
+    def initialize(result_set, action, object)
       details = {
-        action: action,
-        object: object,
-        failed_targets: resultset.error_set.names
+        'action' => action,
+        'object' =>  object,
+        # CODEREVIEW these aren't target objects should we remove this?
+        'failed_targets' => result_set.error_set.names,
```

Comment:
Why not pass the target objects?

Reply:
this was here previously to avoid putting the entire result_set in details. I'm not sure saving a single method call is worth the duplication